server/discussionAnchor/model.ts

```diff
@@ -0,0 +1,18 @@
+export default (sequelize, dataTypes) => {
+	return sequelize.define(
+		'DiscussionAnchor',
+		{
+			id: sequelize.idType,
+			isOriginal: { type: dataTypes.BOOLEAN, allowNull: false },
+			discussionId: { type: dataTypes.UUID, allowNull: false },
+			historyKey: { type: dataTypes.INTEGER, allowNull: false },
+			selection: { type: dataTypes.JSONB, allowNull: true },
```

Comment:
Quick check just to verify my mental model how this is working:

- We used to store this `selection` value in Firebase. When the document changed, one of the clients would write a new value to the `selection` field that matched the `currentKey` of the doc.
- Each client manually managed all their local discussion selections though - so that value would only be used for new pageloads, not to keep decorations in sync across client instances.
- We now store this information in postgres, and update it at a similar cadence, and again, it's only relevant for page loads.

I think I'm following from the code below, but just to be explicit, what is the intention of when this field gets updated, and who triggers it?

Reply:
This doesn't replace the Firebase discussion anchoring information — just the `Anchors` table that currently exists. The primary difference between that table and this one is that the `selection` field for a given `DiscussionAnchor` never gets updated — instead, when a Release is created, we create a new `DiscussionAnchor` with a future history key that associates it with that Release. Keeping the old one around will help us anchor comments on previous Releases.